pkg/cmd/grafana-cli/commands/install_command_test.go

```diff
@@ -228,13 +220,17 @@ func setupPluginInstallCmd(t *testing.T, pluginDir string) utils.CommandLine {
 		return plugin, nil
 	}
 
-	client.DownloadFileFunc = func(pluginName, filePath, url string, checksum string) (content []byte, err error) {
+	client.DownloadFileFunc = func(pluginName string, tmpFile *os.File, url string, checksum string) (err error) {
 		assert.Equal(t, "test-plugin-panel", pluginName)
 		assert.Equal(t, "/test-plugin-panel/versions/1.0.0/download", url)
 		assert.Equal(t, "test", checksum)
-		body, err := ioutil.ReadFile("testdata/grafana-simple-json-datasource-ec18fa4da8096a952608a7e4c7782b4260b41bcf.zip")
+		f, err := os.Open("testdata/grafana-simple-json-datasource-ec18fa4da8096a952608a7e4c7782b4260b41bcf.zip")
+		assert.Nil(t, err)
+		_, err = io.Copy(tmpFile, f)
+		assert.Nil(t, err)
+		err = tmpFile.Close()
```

Comment:
Why is it this function's responsibility to close `tmpFile`, shouldn't that be done by the function that opens it?

Reply:
Ok, I've changed that.